### DISCUSSION TITLE
Select a mesh body using mouse click

### DISCUSSION THREAD

**Discussion by venugovh** [pyvista/pyvista#2140] [Q&A] [answered]
Hi,

Is there a way to select a complete mesh from a `pyvista` plotter window with multiple mesh bodies using a mouse click? I know the cell picking enables the user to drag a box and record the selection, but this would be quite daunting if the meshes are clustered together closely.

The `pyvistaqt.BackgroundPlotter` has a similar method to select and disable a certain vtk object, but would it possible to do that with a mouse click?

Any help will be really appreciated. Thanks!

**Comment by banesullivan** [comment, marked as answer]
You can do this with the `use_mesh` option of `enable_point_picking`:

```py
import pyvista as pv

class Picker:
    def __init__(self):
        self.mesh = None
    def __call__(self, picked, *args):
        self.mesh = picked
        
picker = Picker()

mesh = pv.Sphere()
p = pv.Plotter(notebook=0)
p.add_mesh(mesh)
p.enable_point_picking(picker, use_mesh=True)
p.show()
```

After picking and closing the plotter:

```py
>>> picker.mesh
PolyData (0x7f93c0fa89f0)
  N Cells:	1680
  N Points:	842
  X Bounds:	-4.993e-01, 4.993e-01
  Y Bounds:	-4.965e-01, 4.965e-01
  Z Bounds:	-5.000e-01, 5.000e-01
  N Arrays:	1
```

**Reply by banesullivan** [reply]
Should work the same for the background plotter

**Reply by venugovh** [reply]
Thank you very much! It works :)
